sponsor-dapp-v2/src/lib/custom-hooks.js

```diff
@@ -57,3 +58,76 @@ export function useFaucetUrls() {
       return {};
   }
 }
+
+export function useTextInput() {
+  const [amount, setAmount] = useState("");
+  const handleChangeAmount = event => {
+    // Check if regex number matches
+    if (/^(\s*|\d+)$/.test(event.target.value)) {
```

Comment:
nit: is this regex decimal-ready? If not, could we add a TODO to add that?

Also, just for my own info, do we limit how large the input can get for these text boxes? If so, is that embedded in the regex or is that does that happen elsewhere?

Reply:
This regex is not decimal-ready. I have that in my next PR.

The input length is limited on the text box itself, not via the regex.